services/src/projects/project.rs

```diff
@@ -53,16 +55,25 @@ impl Project {
             project.description = description;
         }
 
-        if let Some(layers) = update.layers {
-            for (i, layer) in layers.into_iter().enumerate() {
-                if let Some(layer) = layer {
-                    if i >= project.layers.len() {
-                        project.layers.push(layer);
-                    } else {
-                        project.layers[i] = layer;
-                    }
+        if let Some(layer_updates) = update.layers {
+            let layers = mem::replace(&mut project.layers, Vec::new()).into_iter();
+            let mut layer_updates = layer_updates.into_iter();
+
+            for (layer, layer_update) in layers.zip(&mut layer_updates) {
+                match layer_update {
+                    LayerUpdate::Keep => project.layers.push(layer),
+                    LayerUpdate::Update(updated_layer) => project.layers.push(updated_layer),
+                    LayerUpdate::Delete => {}
                 }
             }
+
+            for layer_update in layer_updates {
+                if let LayerUpdate::Update(new_layer) = layer_update {
+                    project.layers.push(new_layer);
+                }
+
+                // TODO: throw errors on other cases?
```

Comment:
yes, please do

Reply:
Currently, you didn't consider it to fail. The functionality is not enhanced by throwing, so should we really do this?

Reply:
it makes it easier to catch bugs in the frontend where it wants to update layers that do not exist.

Reply:
https://github.com/geo-engine/geoengine/pull/115/commits/ec3fb0f6bac6bbf3e76339fc1ee4d2bf0c6f7c13#diff-81d6c97ee571fac79cd18261a47f66164451154aa8f637d1b60cbc9dacea210aR80